### ISSUE TITLE
Fix empty links in /browse list

### ISSUE THREAD

**Issue by erinchambers** [CDRH/orchid#293]
Linked "....." inside `<td class="index_spacer">`  doesn't meet [WCAG 2.4.4](https://webaim.org/standards/wcag/checklist#sc2.4.4). Some options for meeting this standard:

- Unlink the `index_spacer`
- Add an `aria-label` to the link
- Provide text within the link
